src/engine/shared.rs

```diff
@@ -292,6 +293,32 @@ pub fn total_allocated(allocated: &Diff<Bytes>, metadata_size: &Diff<Bytes>) ->
     }
 }
 
+/// Convert a u64 value representing seconds, and an optional u32 value
+/// representing nanoseconds to a timestamp.
+pub fn unsigned_to_timestamp(secs: u64, nanos: Option<u32>) -> StratisResult<DateTime<Utc>> {
```

Comment:
I'm usually not a big supported of multiple values evaluating to the same input when you can help it. For example, `0` and `None` are identical from an input perspective, so I'm not seeing what the value of the `Option` input is here.

Reply:
Simplified!